### DISCUSSION TITLE
I cannot see Graph or any traffic animation in Kiali

### DISCUSSION THREAD

**Discussion by zeynalhajili** [kiali/kiali#6906] [Q&A] [answered]
Hey team, 

We have already istio running in the production environment and we have our own Grafana/Prometheus deployment which is installed via helm charts. My task is to check and analyze current Istio deployment and make improvements for the future. I have realized that we have only istiod pod running in istio-system namespace, so I started with installing the Kiali deployment. Per the documentation, only Prometheus integration is enough to collect metrics for populating the graph, since we are not using Grafana/Prometheus deployment from Istio itself I understand that I need to manually update CM of Kiali and restart deployment with the current service name of Prometheus like below.

`spec:
  external_services:
    prometheus:
      # Prometheus service name is "metrics" and is in the "telemetry" namespace
      url: "http://metrics.telemetry:9090/"
`
I already did it and I restarted deployment, I dont see any logs in Kiali pod regarding Prometheus and I can see correct version of it from Kiali dashboard. I can see only logs regarding tracing which I believe I can skip for now. 

I also went to Prometheus console and tried to use istio_requests_total in expression, but did not get anything as result. Could you please help me to check what could be the issue?

**Comment by jmazzitelli** [comment]
(moving to Q&A discussion, not a bug)

Please see the [FAQ "Why is my graph empty?" here.](https://kiali.io/docs/faq/graph/#emptygraph)

**Reply by zeynalhajili** [reply]
Thanks for your answer. I have read the document I can say following. 

a) I have checked 3 namespaces out of 10 which istio-injection is enabled, but nothing.
b) I have sidecars injected successfully, the only issue I think is with Prometheus since as I stated we do have our Prometheus,so I am not installing Prometheus which is in the addons directory. Any idea for troubleshooting?

**Reply by jmazzitelli** [reply, marked as answer]
This would be a question for the Istio folks... sounds like you do not have Istio configured correctly for getting Prometheus to collect and store telemetry from the Istio sidecars. Check the Istio docs for how to configure Prometheus in order to collect Istio telemetry.

Once you see that `istio_requests_total` data get into Prometheus, then you should see the graph. But until Istio is configured properly to get your Prometheus to scrape the data from the proxies, then Kiali won't be able to show you the graph.

**Reply by zeynalhajili** [reply]
Thanks a lot, can I move this question there please if you don't mind?

**Reply by jmazzitelli** [reply]
You cannot move issues/discussions across different github organizations -- this Q&A discussion is in the github "kiali" organization, you want to ask in the istio organization. But I don't think the Istio folks use github for community questions. Please see https://istio.io/latest/get-involved/ -- the first section "Ask questions"

**Reply by jmazzitelli** [reply]
We typically engage with the Istio community on [slack](https://slack.istio.io/) - that is mentioned there in that link I just posted.

**Reply by nrfox** [reply]
@hacizeynal as a place to start, I would check if your workload pods show up as "Targets" in Prometheus. You could also compare the addon deployment of Prometheus to your deployment and see if there's any difference there. Specifically comparing the scrape config would probably be useful.

**Reply by zeynalhajili** [reply]
I have checked targets and did not find anything related to istio, but in my lab, I can see them since there I am using Prometheus from Istio. 

<img width="1263" alt="Screenshot 2023-12-04 at 10 55 09" src="https://github.com/kiali/kiali/assets/38220178/5a04bbb4-1c85-4abc-be74-6b996ff3bd8f">

**Reply by jmazzitelli** [reply]
Which confirms what I think - you have an Istio misconfiguration. You need to tell Istio to work with your own Prometheus, but somewhere that configuration is wrong. Talk to the Istio folks as we are not the true experts on configuring the Istio telemetry setup.

**Reply by zeynalhajili** [reply]
Thanks @jmazzitelli :)

**Comment by nrfox** [comment]
> I also went to Prometheus console and tried to use istio_requests_total in expression, but did not get anything as result. Could you please help me to check what could be the issue?

If there's no metrics in prometheus then you won't see the graph in Kiali since Kiali uses prom metrics to generate the graph. It seems like a prometheus configuration issue. Check out the link @jmazzitelli posted above and maybe check that your pods are being scraped properly.

**Reply by zeynalhajili** [reply]
Hello team,
I fixed the issue with Prometheus and I can see now my pods when I go to targets. I have 4 namespaces with istio, one for mirroring task and other 3 for mTLS lab. Interestingly for my mirroring lab, I can see a graph populated in Kiali and I can see traffic animation, but on the other hand, for my mTLS lab, I do not see anything. I am doing this [lab](https://istio.io/latest/docs/tasks/security/authentication/mtls-migration/), with changing namespaces and I am sending request with following command and I can see successful 200 codes

`while true; do for from in "protected-area1" "protected-area1" "unprotected-area1"; do for to in "protected-area1" "protected-area2"; do kubectl exec "$(kubectl get pod -l app=sleep -n ${from} -o jsonpath={.items..metadata.name})" -c sleep -n ${from} -- curl http://httpbin.${to}:8000/ip -s -o /dev/null -w "sleep.${from} to httpbin.${to}: %{http_code}\n"; done; done; sleep 5; done
`
